core/internal/gethwrappers/go_generate_test.go

```diff
@@ -131,35 +84,12 @@ func init() { // compute rootDir
 func compareCurrentCompilerArtifactAgainstRecordsAndSoliditySources(
 	t *testing.T, versionInfo ContractVersion,
 ) {
-	apath := versionInfo.CompilerArtifactPath
-	contract, err := ExtractContractDetails(apath)
-	require.NoError(t, err, "could not get details for contract %s", versionInfo)
-	hash := contract.VersionHash()
+	hash := VersionHash(versionInfo.AbiPath, versionInfo.BinaryPath)
 	recompileCommand := fmt.Sprintf("(cd %s; make go-solidity-wrappers)", rootDir)
 	assert.Equal(t, versionInfo.Hash, hash,
-		boxOutput(`compiler artifact %s has changed; please rerun
+		boxOutput(`compiled %s and/or %s has changed; please rerun
 %s,
-and commit the changes`, apath, recompileCommand))
-
-	// Check that each of the contract source codes hasn't changed
-	soliditySourceRoot := filepath.Dir(filepath.Dir(filepath.Dir(apath)))
-	contractPath := filepath.Join(soliditySourceRoot, "src", filepath.Base(filepath.Dir(versionInfo.CompilerArtifactPath)))
```

Comment:
How do we check that the golang wrappers are built from ABI outputs which are compiled from the latest solidity contracts, without this test?

Reply:
This test is still in here on line [89](https://github.com/smartcontractkit/chainlink/pull/4010/files/e1d2dc91afe1f7f1ed8c73d004291fd2bdb05f5b#diff-5d7f61d0e18d0ea69b95b3ae8e2e9dce880b69d9c5ba268f06e9f7cfc187e710L139).

It still generates abis and bins, and checks them against the current wrappers, but it doesn't check the "Sources" because that doesn't exist now that we're not using JSON artifacts from belt.

Reply:
I see. I think we still need some way to check that the ABI/binary files correspond to the current version of the solidity sources. Maybe we could add a hash of the solidity source files to `VersionHash`?

Reply:
@kaleofduty suggests that we could add a check for this in CI: build the ABI/binaries, and check that they match byte-for-byte. What do you think?

Reply:
+1 for a CI check.

Reply:
I added a CI check